CMakeLists.txt

```diff
@@ -145,6 +145,35 @@ else()
   # and it is not obvious that there is a demand to do this currenly.
 endif()
 
+# Enable the inclusion of the Discord "Rich Presence" RPC support library and
+# our interface code. The latter is included in the binary code by default
+# unless the environmental variable WITH_DISCORD is defined AND is set to a
+# (case insensitive) value of "NO". Linux packagers for 32-bit targets will find
+# it useful to do this since Discord is not supported on that target and it
+# has proven hard to detect the target's bitness for both QMake and CMake build
+# systems. FreeBSD is also not supported but that is handled automatically.
+# WITH_DISCORD is an environmental value/variable (so could be a number, a
+# string, something else or not even exist).
+set(DISCORD_TEST $ENV{WITH_DISCORD})
+if((CMAKE_SYSTEM_NAME STREQUAL "Linux") OR (CMAKE_SYSTEM_NAME STREQUAL "Windows") OR (CMAKE_SYSTEM_NAME STREQUAL "Darwin"))
```

Comment:
I don't want this additional complication in the Makefile and everywhere in the code, I made the Discord support be automatically picked up for this reason - if it works, it'll work, and if it doesn't work, it gets turned off.

Reply:
What header file should a FreeBSD build include?

OK, I'll take another look to see if I can get back to just the dynamic loading, and have a look as to why it wasn't working for me on that OS.